docs/source/plugins/optional/multiplexer.rst

```diff
@@ -0,0 +1,194 @@
+    ┗━━ os
+         ┣━━ distro
+         ┃    ┗━━ redhat
+         ┃         ╠══ fedora
+         ┃         ║    ┣━━ version
+         ┃         ║    ┃    ╠══ 20
+         ┃         ║    ┃    ╚══ 21
+         ┃         ║    ┗━━ flavor
+         ┃         ║         ╠══ workstation
+         ┃         ║         ╚══ cloud
+         ┃         ╚══ rhel
+         ┃              ╠══ 5
+         ┃              ╚══ 6
+         ┗━━ arch
+              ╠══ i386
+              ╚══ x86_64
+
+Which produces::
+
+   Variant 1:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/workstation, /os/arch/i386
+   Variant 2:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/workstation, /os/arch/x86_64
+   Variant 3:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/cloud, /os/arch/i386
+   Variant 4:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/cloud, /os/arch/x86_64
+   Variant 5:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/workstation, /os/arch/i386
+   Variant 6:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/workstation, /os/arch/x86_64
+   Variant 7:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/cloud, /os/arch/i386
+   Variant 8:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/cloud, /os/arch/x86_64
+   Variant 9:    /os/distro/redhat/rhel/5, /os/arch/i386
+   Variant 10:    /os/distro/redhat/rhel/5, /os/arch/x86_64
+   Variant 11:    /os/distro/redhat/rhel/6, /os/arch/i386
+   Variant 12:    /os/distro/redhat/rhel/6, /os/arch/x86_64
+
+Versus Jenkin's sparse matrix::
+
+   os_version = fedora20 fedora21 rhel5 rhel6
+   os_flavor = none workstation cloud
+   arch = i386 x86_64
+
+   filter = ((os_version == "rhel5").implies(os_flavor == "none") &&
+             (os_version == "rhel6").implies(os_flavor == "none")) &&
+            !(os_version == "fedora20" && os_flavor == "none") &&
+            !(os_version == "fedora21" && os_flavor == "none")
+
+Which is still relatively simple example, but it grows dramatically with
+inner-dependencies.
+
+MuxPlugin
+~~~~~~~~~
+
+:class:`avocado_varianter_yaml_to_mux.mux.MuxPlugin`
+
+Defines the full interface required by
+:class:`avocado.core.plugin_interfaces.Varianter`. The plugin writer
+should inherit from this ``MuxPlugin``, then from the ``Varianter``
+and call the::
+
+   self.initialize_mux(root, paths, debug)
+
+Where:
+
+* root - is the root of your params tree (compound of :ref:`tree-node` -like
+  nodes)
+* paths - is the :ref:`parameter-paths` to be used in test with all variants
+* debug - whether to use debug mode (requires the passed tree to be
+  compound of ``TreeNodeDebug``-like nodes which stores the origin
+  of the variant/value/environment as the value for listing purposes
+  and is __NOT__ intended for test execution.
+
+This method must be called before the :ref:`varainater`'s second stage
```

Comment:
The reference is correct, but the original anchor has a typo (s/varainater/varianter/)